### PR TITLE
Remove hardcoded plugin extension from RUN command

### PR DESCRIPTION
**Pull request by w0land**
On OS X shared object extension is dylib hence test invocation had to be
changed to reflect that.
This will probably fix this issues on other platforms (Windows?)

### CODE REVIEW

File: test/exclude-system-headers.cpp [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// RUN: %clang -fsyntax-only -std=c++11 -w -ferror-limit=0 -Xclang -verify -Xclang -load -Xclang %llvmshlibdir/misracpp2008.so -Xclang -plugin -Xclang misra.cpp.2008 -Xclang -plugin-arg-misra.cpp.2008 -Xclang all %s
+// RUN: %clang -isystem -fsyntax-only -std=c++11 -w -ferror-limit=0 -Xclang -verify -Xclang -load -Xclang %llvmshlibdir/misracpp2008%pluginext -Xclang -plugin -Xclang misra.cpp.2008 -Xclang -plugin-arg-misra.cpp.2008 -Xclang all %s
```
Comment (rettichschnidi) [inline test/exclude-system-headers.cpp]: This kills the tests on my local machine. Why exactly do you need the "isystem" flag?

To me it looks like an error that swallows the -fsyntax-only option.

**Comment by rettichschnidi** [comment]
Thanks for the patch! Will happly accept once the isystem issue can be resolved.

**Comment by w0land** [comment]
Ah, this shouldn't be here. 
I was having problems with finding standard headers, like cassert, during clang invocation (tests were failing with message 'casssert' no such file)
It turns out on OS X you also need libcxx from llvm to correctly find C++ headers (clang from local out of source build is not using system headers). 
tl;dr This isystem stuff is not needed plus on OS X you need libcxx. I'll adopt the patch later today
